modules/tinymce/src/core/test/ts/browser/DragDropOverridesTest.ts

```diff
@@ -8,194 +8,299 @@ import { assert } from 'chai';
+
+      const toolbar = UiFinder.findIn(SugarBody.body(), '.tox-toolbar__primary').getOrDie();
+      await DragnDrop.pDropFiles(toolbar, [
+        createFile('test.js', 123, new Blob([ 'var a = "content";' ], { type: 'application/javascript' }))
+      ]);
+      await pAssertNotification('Dropped file type is not supported');
+    });
+
+    it('TINY-8874: Dragging CEF element towards the bottom edge causes scrolling', async () => {
+      const editor = hook.editor();
+      editor.setContent(`
       <p contenteditable="false" style="height: 200px; background-color: black; color: white">Draggable CEF</p>
       <p id="separator" style="height: 100px"></p>
       <p id="hidden" style="height: 300px"></p>
       <p>CEF can get dragged after this one</p>
     `);
-    const target = UiFinder.findIn(TinyDom.body(editor), 'p:contains("Draggable CEF")').getOrDie();
-    const dest = UiFinder.findIn(TinyDom.body(editor), '#separator').getOrDie();
-    const initialScrollY = editor.getWin().scrollY;
-    Mouse.mouseDown(target);
-    Mouse.mouseMoveTo(dest, 0, 98); // Move the mouse close to the bottom edge of the editor to trigger scrolling
-    await Waiter.pWait(1000); // Wait a small amount of time to ensure the scrolling happens
-    Mouse.mouseUp(target);
-    assert.isTrue(editor.getWin().scrollY > initialScrollY); // Make sure scrolling happened
-  });
+      const target = UiFinder.findIn(TinyDom.body(editor), 'p:contains("Draggable CEF")').getOrDie();
+      const dest = UiFinder.findIn(TinyDom.body(editor), '#separator').getOrDie();
+      const initialScrollY = editor.getWin().scrollY;
+      Mouse.mouseDown(target);
+      Mouse.mouseMoveTo(dest, 0, 98); // Move the mouse close to the bottom edge of the editor to trigger scrolling
+      await Waiter.pTryUntil('Waiting for the editor to scroll down', () => {
+        assert.isAbove(editor.getWin().scrollY, initialScrollY); // Make sure scrolling happened
+      });
+    });
 
-  it('TINY-8874: Dragging CEF element towards the upper edge causes scrolling', async () => {
-    const editor = hook.editor();
-    editor.setContent(`
+    it('TINY-8874: Dragging CEF element towards the upper edge causes scrolling', async () => {
+      const editor = hook.editor();
+      editor.setContent(`
       <p>CEF can get dragged before this one</p>
       <p id="hidden" style="height: 300px"></p>
-      <p id="separator" style="height: 100px"></p>
+      <p class="separator" style="height: 100px"></p>
       <p contenteditable="false" style="height: 200px; background-color: black; color: white">Draggable CEF</p>
     `);
-    editor.getWin().scroll({
-      top: 450
+      editor.getWin().scroll({
+        top: 400,
+        behavior: 'auto'
+      });
+      await Waiter.pWait(1500);
+      const target = UiFinder.findIn(TinyDom.body(editor), 'p:contains("Draggable CEF")').getOrDie();
+      const initialScrollY = editor.getWin().scrollY;
+      const dest = SugarElement.fromDom(editor.getBody());
```

Comment:
This and other places below should be using the `TinyDom.body(editor)` McAgar helper instead.

Reply:
Done!